### ISSUE TITLE
Local language app and folder names not recognised after update

### ISSUE THREAD

**Issue by abcdaniel** [quicksilver/Quicksilver#2138]
After my two latest updates of QS, to 1.3.0 and 1.3.2 (missed 1.3.1), QS stopped recognising local language app and folder names. For example, trying to open "Anteckningar" (Swedish) by typing "ant" doesn't work, instead I have to type "not" to trigger "Notes".

I have for several years had all commands in QS in English (Terminal: defaults write com.blacktree.Quicksilver AppleLanguages -array English) and up until now QS has still recognised the local names for apps and folders as they appear on my system.

I have reverted the behavior, by quitting QS, clearing caches, Terminal: defaults write com.blacktree.Quicksilver AppleLanguages -array Swedish, starting QS, quitting QS, Terminal: defaults write com.blacktree.Quicksilver AppleLanguages -array English.

OS X 10.9.5, QS 1.3.2

**Comment by tiennou** [comment]
QS clears caches when it updates — that would revert your "hack". I'm not sure I'd be supporting such a mixed environment though...

**Comment by abcdaniel** [comment]
Well, it is not a hack, it is the official way to make QS work in English on a localised system. From the Wiki:
http://qsapp.com/wiki/FAQ#I.27m_used_to_Quicksilver_in_English_and_don.27t_want_it_translated_into_my_language._How_do_I_make_that_happen.3F

It is a great feature, most pro apps are not translated to Swedish, so English commands come natural + not all plug-ins are translated, creating a situation where you have to switch language midway some commands if having QS localised. Not very intuitive.

The solution to having QS in English, as described in the Wiki, has worked fine for several years, with the system remaining interpreted in Swedish. It only now got broken in 1.3.0.

**Comment by tiennou** [comment]
There's something I don't get. The Wiki describes a way to make QS English-*only*. What I described as "hack" above makes Quicksilver get the localized names in the Catalog *once* (when you run with Swedish), but on the next English-only rescan it will be replaced, which is why I dubbed it as a hack — that will only work for like 10 minutes.

Changing the `-AppleLanguages` default (as per the FAQ) overrides the OS language preferences that comes from the list of enabled languages in "System Preferences > Regional Settings". So it's normal that the Swedish localized filenames don't appear if you do, because the OS never give them to us when we ask for the localized name of e.g. Desktop. If you want Swedish *and* English file names at the same time, you should not use the "hack" and use the default.

The only thing that ought to have changed with 1.3.0 is that someone translated the actions to Swedish, those translations were pulled in, and they now get selected when displaying localized actions names.

**Comment by abcdaniel** [comment]
Maybe we are misunderstanding one another? I'm not a programmer, so pardon any ignorance on my part. 

QS defaults to local language for its interface, actions and commands, which I don't want. I once a long time ago applied the Wiki instructions. The result: QS, interface, actions and commands, fully in English, but it also recognises localised folder and app names, like Skrivbord (Desktop), Bibliotek (Library), Bilder (Pictures).

This was persistant through several years and updates. I consider it well behaved; I ask QS to be in English, not to interpret my filesystem in English. It certainly does not break on rescan. 

Thing is since 1.3.0 it breaks on updating QS, and I have to do the process I outlined above again. It is then well behaved again and this persists until the next update. It does not break on rescan.

**Comment by tiennou** [comment]
No worries, and let's hope I don't come as too full of myself ;-).

The thing that bothers me is that I believe the FAQ entry doesn't really do what you want it to. Let me explain : the underlying filesystem names are actually always in English (which you can see using Terminal.app), and are translated by the localization machinery when we ask for the localized name — mostly through `NSURLLocalizedNameKey`. So when you say `-AppleLanguages something`, I expect this call to return the localized name in the overridden language, not the actual, operating-system defined language one (otherwise the behavior is different when running with this kind of override from when you actually have this set at the system level, which just feels weird to me). So the FAQ entry is only correct if you want only English to ever show up.

Looking a little at the development history, I can see 3feaf36fa178b074aeed6853821f4d726e1fa096, which was merged a few days prior to the release of 1.3.0 (4b20058f3c5e37635b3264f9e187c791d0e67b26), and which seems to corroborate what I explained above : the newly-done translation for Swedish was pulled in at that time, and it only got ~~worse~~better from here, as more and more strings would get translated.

Aaanyway, there is currently no easy way to do what you ask, short of partly short-circuiting the localization mechanism some times (for objects), but not others (for actions). And I can tell you that having "Large Type" translated as "Big Font" bothers me too ;-).

If you really want to run QS the way you describe, you'll have to delete all the `sv.lproj` folders you can find in QS and plugins : this will still allow the OS-provided localization to work while nothing else will be translated (the default locale for most applications being English).

Here's a set of Terminal commands that should do that trick :
```
# Revert the FAQ
defaults delete com.blacktree.Quicksilver AppleLanguages`

# Delete all sv.lproj folders in QS
find /Applications/Quicksilver.app -name sv.lproj -delete

# Delete all sv.lproj in installed plugins
find ~/Library/Application\ Support/Quicksilver/PlugIns -name sv.lproj -delete
```
(Please be careful with `find`, if you want to be sure you don't catch some unwanted thing, replace `-delete` with `-print`)

Doing this will break code-signing though, so you'll get a "not signed" warning on startup, and maybe a Keychain authorization if you have the plugin installed (I know there's a way to manually resign some application but my Google-fu fails me).

**Comment by abcdaniel** [comment]
Thanks for your thorough reply! I think I get it now from a programming/data point of view.
Thing is, the solution I've outlined above still works, is persistant on restart and rescan, and makes QS overall sweet to use. English everything in QS, filesystem using local names.

Maybe this should be a feature request instead of a bug report? I obviously find it "the right way" to have English QS read local file names, but I can surely see the benefit/workflow for some having QS be all out English, incl English file names. A tickbox to rule us all!

Should I file a feature request? Is it possible/desirable to implement?
